### ISSUE TITLE
Portal files aren't deleted filling up the HD

### ISSUE THREAD

**Issue by MalcolmMackenzie** [open-automation/switch-portals#20]
Hi
On a customer's Switch Server the Portal files aren't deleted and it's filling up the HD.
I've checked the folder for the permissions which are full R/W.

Time,Id,Type,Module,Ticket,Flow,Element,File,Message
2017-11-20T09:01:51.804Z,2213273,warning,Outgoing,,INPUT,Cerm Thumb Incoming,,unpackJob returned false. Could not unpack. Skipping this file.
2017-11-20T09:01:51.804Z,2213272,error,Outgoing,,INPUT,Cerm Thumb Incoming,,Unarchiving of the job to the temporary folder was unsuccessful.

Any pointers please to fix this please?
Thanks 
Malcolm Mackenzie

Folder location
C:\Users\ArtworkUser\AppData\Roaming\Enfocus\Switch Server\ScriptData\SwitchPortalsEther\Portals_C-MessageFileUploaded
![screenshot 2017-11-20 10 51 07](https://user-images.githubusercontent.com/25176531/33015514-013802d4-cde3-11e7-9be2-bab750dfbc60.png)

**Comment by Dominick-Peluso-Bose** [comment]
Hi Malcolm, 

This is the method the script is using:

> unarchive( source-path : String, destination-path : String, passwords : String[ ] ) : Boolean
> Extracts all files from the archive at the source path and places them as a file or folder at the destination path. The last (optional) argument provides a list of passwords for opening password-protected archives.
> 
> Returns true if successful, false if not (in which case an appropriate error message is logged).
> 
> This function supports the ZIP and RAR archive formats.
> 
> Very large archives (+4 GB) can be processed as well.
> 
> Note: The first correct password is used for all of the password protected files. If there are files with another password, those files will be skipped.

From here: http://www.enfocus.com/manuals/DeveloperGuide/SW/17/en-us/common/sw/reference/r_environment_class.html

It says that an error message should be logged when _unarchive_ fails. Could you turn on debug log messaging and see what is printed to the log?

**Comment by MalcolmMackenzie** [comment]
Hi Dominick

I'll get the debug going and send through the results.

 Thanks

Malcolm Mackenzie

 07910 413421

**Comment by MalcolmMackenzie** [comment]
I've attached a log file to see if that will help
Thanks
[logs.csv.zip](https://github.com/open-automation/switch-portals/files/1501365/logs.csv.zip)

**Comment by MalcolmMackenzie** [comment]
Any thoughts on next steps please?

**Comment by matthewkayy** [comment]
Hi

I think we have a similar issue with this, if we have a flow turned off with an Outgoing portal when the flow is turned back on it won't pickup any jobs that have been sent into it while its been off. The jobs will just stay in the SwitchPortalsEther folder.

Is this something we're doing wrong with Portals?

Thanks

Matthew

**Comment by Dominick-Peluso-Bose** [comment]
@matthewkayy and the same configuration works when the flow is turned off? If not, then I would think your scope:channel are mismatched. Outgoing Portals in offline flows should process as soon as it goes online again.

**Comment by Dominick-Peluso-Bose** [comment]
@MalcolmMackenzie This is the part that is failing for you: https://github.com/open-automation/switch-portals/blob/master/Unpacked/Outgoing_v3/Outgoing_v3.js#L224-L225

Portals needs to create a temporary folder and unpack the packed job (jobs are packed in zips with metadata between portals) into that temporary folder. After that is complete, a new job is created, all files added back to the job, and metadata is attached to the job. All of this stems from the temporary folder. The _createPathWithName_ method, according to the documentation, should throw an exception if it fails -- so we can assume it worked fine since it got far enough to print the log message below.

So the problem appears to be one of two things, both stemming from Switch's _unarchive_ method.
-  _unarchive_ may not be able to read the contents of the packed zip file
-  _unarchive_ may not be able to write to the temporary folder

The temporary folder is created with http://www.enfocus.com/manuals/DeveloperGuide/SW/17/en-us/common/sw/reference/r_job_class.html So there could be something wrong with your Switch configuration where Switch is not actually giving you a writable path. This method is supposed to do that, but if Switch had some odd permissions, perhaps it could hand out a path to a location it couldn't write to. The best way to test this would be to write a script that would use this method and  write to the temporary path that was created.

The other option is that _unarchive_ couldn't read the incoming file and failed as a result. This could happen if you're using mismatched versions of Portals, because at some point I think I may have changed the naming of packed files between versions. It could also be caused by another permissions issue where Switch, for whatever reason, didn't have permission to read the packed file.

Long story short, I don't exactly know what the issue is. But to troubleshoot, what I would do is to create some new flows, drop in new scripting elements into the flow canvas, add new portals v3 (found here https://github.com/open-automation/switch-portals/tree/master/Packed). If that didn't resolve it, then I would modify the scripts to log each of the possible values above (the path of the temporary directory, the filename unarchive is trying to unzip) etc... until it was apparent. So, do you have the Scripting module?

**Comment by MalcolmMackenzie** [comment]
OK thanks Dominick, we'll have a look
